src/resources/chain.rs

```diff
@@ -280,6 +283,232 @@ impl<T: Block> Chain<T> {
         Err(unsafe { OwnedChain::from_vec_unchecked(res) })
     }
 
+    /// Returns an OwnedChain<T> containing the difference, i.e. the
+    /// values that are in `self` but not in `other`.
+    pub fn difference<C: AsRef<Chain<T>>>(
+        &self, other: &C
+    ) -> OwnedChain<T> {
+        let other = other.as_ref();
+        let mut res: Vec<T> = vec![];
+        
+        // Border case: if self is empty, then the result is an empty OwnedChain.
+        if self.is_empty() {
+            return OwnedChain::empty();
+        }
+        // In this case we need to walk over both chains
+        else {
```

Comment:
You don’t need the else here since you returned in the other branch. Saves you one indentation.